### PR TITLE
Cast value to text to make filter work with uuid primary key columns

### PR DESCRIPTION
**Pull request by vesterbaek**
Check for uuid primary key and add typecast if required to support join between model and watson tables

### CODE REVIEW

**Comment by vesterbaek** [comment]
I'm not sure if the `try` / `except NameError` approach is how you would normally ensure backwards compatibility

File: src/watson/backends.py [hunk 1]
```diff
@@ -274,7 +279,8 @@ def do_filter(self, engine_slug, queryset, search_text):
                 "watson_searchentry.search_tsv @@ to_tsquery('{search_config}', %s)".format(
                     search_config = self.search_config
                 ),
-                "watson_searchentry.{ref_name} = {table_name}.{pk_name}".format(
+                "watson_searchentry.{ref_name}{ref_name_typecast} = {table_name}.{pk_name}".format(
```
Comment (etianen) [inline src/watson/backends.py]: Interesting. I wonder if we couldn't make it more generic by casting `{table_name}.{pk_name}` to string if it's not an integer primary key. That would presumably account more more fields than a UUID.

**Comment by vesterbaek** [comment]
Yup, I actually did that in my first implementation. The title of this pull request is a hint about that :). I added explicit support for the uuid type for performance reasons. We could add a fallback to text cast after the uuid check

**Comment by etianen** [comment]
How does the uuid conversion improve performance over the string
conversion? I'd have thought that both blow the use of indices out of the
water.

**Comment by vesterbaek** [comment]
uuid is a native type with 128 bit length: http://www.postgresql.org/docs/9.1/static/datatype-uuid.html. Representing the uuid as text leads takes more storage (see eg. http://stackoverflow.com/questions/33836749/postgresql-using-uuid-vs-text-as-primary-key). In this case where one table has text and one has uuid, my knowledge of PostgreSQL is not good enough and I'm not sure if there is a performance gain by casting to uuid.

**Comment by etianen** [comment]
I think it's the sort of thing that could only be shown using benchmarks. I
suspect, however, that any performance difference would be negligible,
since the database won't be able to use indices either way. What's certain,
however, is that the code required for the "cast to string" approach is
much smaller, which is a definite win!

**Comment by vesterbaek** [comment]
Closing this one on favor of https://github.com/etianen/django-watson/pull/169

**Pull request closed without merging**